### PR TITLE
[ZEPPELIN-4331] Use latest version of Java driver for Cassandra

### PR DESCRIPTION
**Pull request by alexott**
### What is this PR for?

This change will allow to get access to latest fixes & improvements implemented in the
Datastax Java driver for Cassandra.

### What type of PR is it?
Improvement

### What is the Jira issue?
* https://issues.apache.org/jira/browse/ZEPPELIN-4331

### How should this be tested?
* Automated build: https://travis-ci.org/alexott/zeppelin/builds/585461796
* Manually tested CQL commands

### CODE REVIEW

**Comment by alexott** [comment]
@doanduyhai would you mind to review this?

**Comment by doanduyhai** [comment]
Looks good to me. Do not forget to modify documentation to update Java driver version there too

**Comment by alexott** [comment]
Good point, will push update slightly later

**Comment by alexott** [comment]
added changelog entry

**Comment by doanduyhai** [comment]
LGTM

File: cassandra/pom.xml [hunk 1]
```diff
@@ -88,8 +88,8 @@
         </dependency>
 
         <dependency>
-            <groupId>net.jpountz.lz4</groupId>
-            <artifactId>lz4</artifactId>
+            <groupId>org.lz4</groupId>
+            <artifactId>lz4-java</artifactId>
```
Comment (felixcheung) [inline cassandra/pom.xml]: we need to update https://github.com/apache/zeppelin/blob/79108419e956d4774f87303d970888a8c46132f8/zeppelin-distribution/src/bin_license/LICENSE#L71

File: docs/interpreter/cassandra.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
----
+z---
```
Comment (felixcheung) [inline docs/interpreter/cassandra.md]: why this change?

Comment (alexott) [reply docs/interpreter/cassandra.md]: Just typo, didn't notice when committing.

File: cassandra/pom.xml (hunk 1 above)
Comment (alexott) [reply cassandra/pom.xml]: done...
Thank you for review

Comment (zjffdu) [inline cassandra/pom.xml]: This is required by cassandra java driver 3.7.2 ?

Comment (alexott) [reply cassandra/pom.xml]: @zjffdu stricktly speaking, these are [optional dependencies](https://docs.datastax.com/en/developer/java-driver/3.7/manual/compression/), but because they were used in `pom.xml` already, I just changed them to match versions used in the driver.

**Comment by zjffdu** [comment]
Thanks @alexott LGTM, will merge if no more comment
